runtime/queries/metricsview_comparision_toplist.go

```diff
@@ -0,0 +1,280 @@
+package queries
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/drivers"
+	"github.com/rilldata/rill/runtime/pkg/pbutil"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type MetricsViewComparisonToplist struct {
+	MetricsViewName     string                                 `json:"metrics_view_name,omitempty"`
+	DimensionName       string                                 `json:"dimension_name,omitempty"`
+	MeasureNames        []string                               `json:"measure_names,omitempty"`
+	BaseTimeStart       *timestamppb.Timestamp                 `json:"base_time_start,omitempty"`
+	BaseTimeEnd         *timestamppb.Timestamp                 `json:"base_time_end,omitempty"`
+	ComparisonTimeStart *timestamppb.Timestamp                 `json:"comparison_time_start,omitempty"`
+	ComparisonTimeEnd   *timestamppb.Timestamp                 `json:"comparison_time_end,omitempty"`
+	Limit               int64                                  `json:"limit,omitempty"`
+	Offset              int64                                  `json:"offset,omitempty"`
+	Sort                []*runtimev1.MetricsViewComparisonSort `json:"sort,omitempty"`
+	Filter              *runtimev1.MetricsViewFilter           `json:"filter,omitempty"`
+
+	Result *runtimev1.MetricsViewComparisonToplistResponse `json:"-"`
+}
+
+var _ runtime.Query = &MetricsViewComparisonToplist{}
+
+func (q *MetricsViewComparisonToplist) Key() string {
+	r, err := json.Marshal(q)
+	if err != nil {
+		panic(err)
+	}
+	return fmt.Sprintf("MetricsViewComparisonToplist:%s", string(r))
+}
+
+func (q *MetricsViewComparisonToplist) Deps() []string {
+	return []string{q.MetricsViewName}
+}
+
+func (q *MetricsViewComparisonToplist) MarshalResult() any {
+	return q.Result
+}
+
+func (q *MetricsViewComparisonToplist) UnmarshalResult(v any) error {
+	res, ok := v.(*runtimev1.MetricsViewComparisonToplistResponse)
+	if !ok {
+		return fmt.Errorf("MetricsViewComparisonToplist: mismatched unmarshal input")
+	}
+	q.Result = res
+	return nil
+}
+
+func (q *MetricsViewComparisonToplist) Resolve(ctx context.Context, rt *runtime.Runtime, instanceID string, priority int) error {
+	olap, err := rt.OLAP(ctx, instanceID)
+	if err != nil {
+		return err
+	}
+
+	if olap.Dialect() != drivers.DialectDuckDB {
+		return fmt.Errorf("not available for dialect '%s'", olap.Dialect())
+	}
+
+	mv, err := lookupMetricsView(ctx, rt, instanceID, q.MetricsViewName)
+	if err != nil {
+		return err
+	}
+
+	if mv.TimeDimension == "" {
+		return fmt.Errorf("metrics view '%s' does not have a time dimension", q.MetricsViewName)
+	}
+
+	if q.BaseTimeStart == nil || q.BaseTimeEnd == nil || q.ComparisonTimeStart == nil || q.ComparisonTimeEnd == nil {
+		return fmt.Errorf("undefined time range for comparison on '%s' metrics view ", q.MetricsViewName)
+	}
```

Comment:
Can we make it work like the normal toplist query if the comparison time range is empty? I.e. just return 0 for all the comparison values if no comparison time range is specified?